ARCHITECTURE.md

```diff
@@ -51,34 +51,40 @@ https://github.com/replicatedhq/kurl-api
 
 The browser UI for creating and viewing kURL installers.
 
-#### Go API Proxy
+#### Typescript API
 
-This API proxies most requests directly to the Typescript API.
-Additionally, this API interacts with the object store and is responsible for assembling and streaming air-gap bundles back to the end-user.
+The API for creating kURL URLs and rendering installers.
+The API accepts a kURL Installer spec and returns a URL including a deterministic hash of the spec for installing a cluster based on that spec, for example https://k8s.kurl.sh/5e61e80.
+The API interacts stores these hashes in its relational database for retrieval.
+Additionally, the API is responsible for resolving the add-on version ("latest" and ".x") and rendering the spec when a script is requested, for example https://k8s.kurl.sh/5e61e80/install.sh or shorthand https://k8s.kurl.sh/5e61e80.
+The API reads the current kURL version from the object storage bucket (https://kurl-sh.s3.amazonaws.com/dist/VERSION) and uses that version to lookup add-on version information (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/supported-versions-gen.json) for spec resolution.
 
-#### Typescript API
+#### Go API Proxy
 
-The API for creating and consuming installers. Interacts with the relational database.
+This API proxies most requests directly to the Typescript API.
+Additionally, this API is responsible for assembling and streaming air-gap bundles back to the end-user.
+Air-gap packages are assembled from individual add-on package archives (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/rook-1.0.4.tar.gz) and streamed as a single archive back to the end user when requested (e.g. https://k8s.kurl.sh/bundle/5e61e80.tar.gz).
 
 #### Relational Database
 
 Stores installer hashes.
```

Comment:
What about add here:

How the hashes are generated? What is the source?
Where we can find the schemas? What is the motivation for ? 
How uses this database? vendor, kurl-API ? What API inside of the kurl-API needs to sql the database?
What are the projects that uses these database? What is stored here? Have this database relation with other projects that are not kURL? 

Something like:

> Stores installer hashes built from .... (code/repo) .... . The database schemas are ... . The installer scripts are stored because .... . 

On top of that, I did some questions about that and @laverya shared that the schemas are into : https://github.com/replicatedhq/vandoor/tree/main/migrations/kustomize/schema

Then, in this case I understand that we would have : https://github.com/replicatedhq/vandoor/blob/main/migrations/kustomize/schema/kurl-installer.yaml
But that seems to create the relation that 1 team have 1 installer schema ID. But that still not clear for me.

Should not each release be associated to an installer release?
Or have we not this association?
What happens if I want to upgrade and I have an replicated/kots app/release installed with an old kURL script?
Should not an app/vendor release be associated to script and not k8s supported versions?

Reply:
vandoor is not a public repo, so we can't really link to it here

Reply:
That is fine. But could we clarify how the data is stored/when and etc? Also, I added the questions here to try take advantage and see if I can learn how the projects are connected.

Reply:
Mainly answered as well closing this one.